core/src/main/java/com/linecorp/armeria/client/ClientFactoryOptions.java

```diff
@@ -411,4 +412,26 @@ public ConnectionPoolListener connectionPoolListener() {
     public MeterRegistry meterRegistry() {
         return get0(ClientFactoryOption.METER_REGISTRY);
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    protected <T extends AbstractOptionValue<?, ?>> T mergeValue(T oldValue, T newValue) {
+        if (oldValue.option() == ClientFactoryOption.CHANNEL_OPTIONS) {
+            final ClientFactoryOptionValue<Map<ChannelOption<?>, Object>> castOldValue =
+                    (ClientFactoryOptionValue<Map<ChannelOption<?>, Object>>) oldValue;
+            final ClientFactoryOptionValue<Map<ChannelOption<?>, Object>> castNewValue =
+                    (ClientFactoryOptionValue<Map<ChannelOption<?>, Object>>) newValue;
+            final ImmutableMap.Builder<ChannelOption<?>, Object> builder =
+                    ImmutableMap.builderWithExpectedSize(castNewValue.value().size());
```

Comment:
- Probably better using `oldValue`'s size + `newValue`'s size? Not always optimal, but expansion is more expensive.
- Could just return the `oldValue` if `newValue` is empty (or `newValue` is `oldValue` is empty.)